python/vision_explanation_methods/DRISE_runner.py

```diff
@@ -130,24 +134,52 @@ def get_drise_saliency_map(
 
     test_image = Image.open(image_open_pointer).convert('RGB')
 
-    detections = model.predict(
-        T.ToTensor()(test_image).unsqueeze(0).to(device))
-
-    saliency_scores = drise.DRISE_saliency(
-        model=model,
-        # Repeated the tensor to test batching
-        image_tensor=T.ToTensor()(test_image).unsqueeze(0).to(device),
-        target_detections=detections,
-        # This is how many masks to run -
-        # more is slower but gives higher quality mask.
-        number_of_masks=nummasks,
-        mask_padding=maskpadding,
-        device=device,
-        # This is the resolution of the random masks.
-        # High resolutions will give finer masks, but more need to be run.
-        mask_res=maskres,
-        verbose=True  # Turns progress bar on/off.
-    )
+    if isinstance(model, MLflowDRiseWrapper):
+        x, y = test_image.size
+        imgio = BytesIO()
+        test_image.save(imgio, format='PNG')
+        img_str = base64.b64encode(imgio.getvalue()).decode('utf8')
+        img_input = pd.DataFrame(
+            data=[[img_str, (y, x)]],
+            columns=['image', "image_size"],
+        )
+
+        detections = model.predict(img_input)
+        saliency_scores = drise.DRISE_saliency_for_mlflow(
+            model=model,
+            # Repeated the tensor to test batching
+            image_tensor=img_input,
+            target_detections=detections,
+            # This is how many masks to run -
+            # more is slower but gives higher quality mask.
+            number_of_masks=nummasks,
+            mask_padding=maskpadding,
+            device=device,
+            # This is the resolution of the random masks.
+            # High resolutions will give finer masks, but more need to be run.
+            mask_res=maskres,
+            verbose=True  # Turns progress bar on/off.
```

Comment:
Considering `detections` and `saliency_scores` are both the last variables in the if-else conditions, should prolly push these out of the conditions.